router/usecase/optimized_routes_test.go

```diff
@@ -652,18 +652,53 @@ func (s *RouterTestSuite) TestGetOptimalQuote_Mainnet() {
 			// Mock router use case.
 			mainnetUseCase := s.SetupRouterAndPoolsUsecase(mainnetState)
 
-			// System under test
-			quote, err := mainnetUseCase.Router.GetOptimalQuote(context.Background(), sdk.NewCoin(tc.tokenInDenom, tc.amountIn), tc.tokenOutDenom)
+			// Setup the quotes to be tested
+			quotes := []func() (domain.Quote, error){
+				func() (domain.Quote, error) {
+					return mainnetUseCase.Router.GetOptimalQuote(context.Background(), sdk.NewCoin(tc.tokenInDenom, tc.amountIn), tc.tokenOutDenom)
+				},
+				func() (domain.Quote, error) {
+					return mainnetUseCase.Router.GetOptimalQuoteInGivenOut(context.Background(), sdk.NewCoin(tc.tokenInDenom, tc.amountIn), tc.tokenOutDenom)
+				},
+			}
```

Comment:
I propose sharing the `tests` slice into the global space but still having separate test fixtures for "in given out" and "out given in" for readability.

Additionally, I don't think the current test for "in given out" is very useful because we essentially provide the same "in given out" quote under the hood.

As a result, for "in given out", I propose also calling:
```
_, _, err = quote.PrepareResult(context.Background(), osmomath.OneDec())
s.Require().NoError(err)
```

So that we can validate the reformatted route

Reply:
I think the goal of `TestPrepareResult` is different.

Here, we are trying to test that "in given out" quote functions off of mainnet state, acting more of a functional test.

To validate that properly, we should call `PrepareResult` because that's what converts "out given in" quote to "in given out"